ui.apps/src/main/content/jcr_root/apps/etoolbox-rollout-manager/clientlibs/rollout-manager-ui/js/rollout-utils.js

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+/**
+ * EToolbox Rollout Manager clientlib.
+ * 'Rollout' button and dialog actions definition.
+ */
+(function ($, ns, Granite) {
+    'use strict';
+
+    const ROLLOUT_COMMAND = '/content/etoolbox/rollout-manager/servlet/rollout';
+    const CHECK_STATUS_COMMAND = '/content/etoolbox/rollout-manager/servlet/rollout/status';
+    const SUCCESS_REPLICATION_MSG = Granite.I18n.get('Rollout is completed. Publishing is in progress.');
+    const SUCCESS_MSG = Granite.I18n.get('Rollout completed');
+    const PROCESSING_ERROR_MSG = Granite.I18n.get('Rollout failed because of');
+    const STATUS_UPDATE_INTERVAL = 5000;
+
+    async function doItemsRollout(data) {
+        const logger = ns.createLoggerDialog();
+        try {
+            const response = await buildRolloutRequest(data);
+            logger.unblocked();
+            if (response.task) {
+                await createStatusUpdater(logger)(response.task);
+                logger.finished(data.shouldActivate ? SUCCESS_REPLICATION_MSG : SUCCESS_MSG);
+            }
+        }  catch(e) {
+            logger.finished(PROCESSING_ERROR_MSG + ' ' + e);
+        }
+    }
+    ns.doItemsRollout = doItemsRollout;
+
+    async function buildRolloutRequest(dialogData) {
+        const data = {
+            _charset_: 'UTF-8',
+            selectionJsonArray: JSON.stringify(dialogData.selectionJsonArray),
+            isDeepRollout: dialogData.isDeepRollout,
+            shouldActivate: dialogData.shouldActivate
+        }
+
+        try {
+            return await $.ajax({
+                url: ROLLOUT_COMMAND,
+                type: 'POST',
+                data
+            });
+        } catch (e) {
+            throw new Error(e.responseJSON.error);
+        }
+    }
+
+    function createStatusUpdater(logger) {
+        return async function(taskId, offset = 0) {
+            try {
+                const response = await getStatusInfo(taskId, offset);
+
+                if (response.error) throw new Error(`${response.error}`);
+
+                if (response.messages && response.messages.length) {
+                    offset = response.messages.reduce((total, msg) => {
+                        logger.log(msg);
+                        return msg.id > total ? msg.id : total;
+                    }, offset);
+                }
+
+                if (!response.status) {
+                    throw new Error('Wrong response from the server');
+                } else if (response.status === 'active') {
+                    await promisifyTimeout(STATUS_UPDATE_INTERVAL);
+                    await createStatusUpdater(logger)(taskId, offset);
+                }
+            } catch (e) {
+               throw e;
+            }
+        }
+    }
+
+    async function getStatusInfo(taskId, offset) {
+        try {
+            const url = `${CHECK_STATUS_COMMAND}?task=${taskId}` + (offset ? `&offset=${offset}` : '');
```

Comment:
It seems a little clearer now
```suggestion
            const params = new URLSearchParams({task: taskId});
            if (offset) params.append('offset', offset);
            const url = `${CHECK_STATUS_COMMAND}?${params}`;
```